LWCOSC/force-app/main/default/classes/ProductController.cls

```diff
@@ -0,0 +1,35 @@
+public with sharing class ProductController {
+    
+    @AuraEnabled(cacheable=true)
+    public static List<Product2> getProducts(String searchKey, String sortBy, String sortDirection) {
+        ProductServiceImpl proSerImpl = new ProductServiceImpl();
```

Comment:
You should not invoke the service impl directly, use reflection instead

Reply:
Done